### PR TITLE
Improve requirements

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~Reset the build number to `0` (if the version changed)~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

The main goal of this PR is to sort out any overlinking-related warnings. The main tool for that is fixing any dependency irks. In the process, this is what came out:

- Enable zip support for zarr to justify dependency on libzip
- Fix Bzip2 plugin installation by a patch to the Cmake build system
- Remove obsolete msinttypes dependency that was only necessary for old vc versions
- Replace `curl` dependency with `libcurl` dependency, which also brings the right run_export
- Add libaec dependency which provides libsz, but was missing so far
- Order deps alphabetically
- Simplify run reqs with better use of run_exports.

It also aligns the Windows and Linux builds more closely by
- Using the same ordering of options in the Cmake line if they are the same
- Adding the following to Windows, which was only in the Linux build up to now
    - `BUILD_UTILITES=ON`
    - `ENABLE_DOXYGEN=OFF`
    - `ENABLE_DAP=ON`
    - `ENABLE_NETCDF_4=ON`
    - `ENABLE_PLUGIN_INSTALL=ON`
- Adding the following to Linux which was only in the Windows build up to now
    - `ENABLE_BYTERANGE=ON`

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -59,30 +59,30 @@ requirements:
...
-    - libjpeg-turbo
-    - libzip
-    - zstd
+    - libaec
+    - libcurl
+    # - libjpeg-turbo
     - libxml2
-    - blosc
+    # - libzip
+    - msinttypes  # [win and vc<14]
```
Comment (xylar) [inline recipe/meta.yaml:81]: I don't think we intend to support `vc<14` so I think this can go.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
Three linker warnings remain:
```
WARNING (libnetcdf): dso library package conda-forge::libstdcxx-ng-13.1.0-hfd8a6a1_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (libnetcdf): run-exports library package conda-forge::libjpeg-turbo-2.1.5.1-h0b41bf4_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (libnetcdf): run-exports library package conda-forge::bzip2-1.0.8-h7f98852_4 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```
- I think the Bzip2 part is related to #179.
- Libjpeg-turbo should be used in the HDF4 part, investigation on why it is not linked is ongoing.
- The cxx standard library is a bit strange. How is CXX used in netcdf-c anyway?

**Comment by zklaus** [comment]
Side note: All Osx builds are currently failing on Conda-forge, see conda-forge/python-libarchive-c-feedstock#35.

**Comment by zklaus** [comment]
Beware of issues flagged in #160 and #166!

**Review by ocefpaf — APPROVED** [review]

**Review by xylar — APPROVED** [review]
I would certainly like someone with more expertise than me to have a look but this looks really excellent to me.

**Review by dopplershift — APPROVED** [review]
Seems generally reasonable to me.

**Comment by ocefpaf** [comment]
Ok, 3 approvals is good enough :-)
Let's get this one merged.

Thanks @zklaus for the awesome work you put into this!

**Comment by zklaus** [comment]
Cheers guys :smiley:

**Comment by valeriupredoi** [comment]
@zklaus is this now compiled with byte-range flag on? (sorry, didn't want to go through all the changed files haha). If so then this means we now have S3 support by my back-of-the-enevolope calculations? Many thanks, bud :beer:

**Comment by zklaus** [comment]
Yes, @valeriupredoi, byte-range is available now. And I think you are right that this is enough for S3 access, particularly for single netcdf4 files. Though as I understand the discussion in #160 and #166, you may need to tell libnetcdf about it by appending something like `#mode=bytes` to the URL.

**Comment by valeriupredoi** [comment]
Ooh nice thanks bud - am gonna reference your comment here to our issue, so I don't forget both the info and the references, it's getting a bit Googla Map-y with all these issues and PRs. Cheers muchly :beer:

**Pull request merged**